src/main/java/com/minecolonies/coremod/entity/ai/minimal/EntityAISleep.java

```diff
@@ -214,9 +215,9 @@ private void findBedAndTryToSleep()
             {
                 final IBuilding hut = colony.getBuildingManager().getBuilding(citizen.getHomePosition());
                 List<BlockPos> bedList = new ArrayList<>();
-                if (hut.hasModule(LivingBuildingModule.class))
+                if (hut.hasModule(BedHandlingModule.class))
```

Comment:
Seeing this structure makes my stumach crinch a bit.
I think we should add something like this to the building:
`
public Optional<T extends IModule> <T> getModule(Class<T> moduleClass)
`

On which we can then to chaining with things like `.map(IModuleWithExternalBlocks::getRegisteredBlocks).orElse(Collections.emptyList())`